doc/arch/adr-001-feature-flags.md

```diff
@@ -0,0 +1,57 @@
+# 001 - Feature Flags
+
+## Context
+
+We need to add a mechanism to toggle features in MMW so that we can deploy
+multiple implementations from the same codebase. For example, in BiG CZ,
+we'll need to disable Analyze mode and other MMW-specific features such as
+ITSI integration.
+
+Primary deciding factors:
+
+* How will this impact development?
+* How easy will it be to switch from MMW to BiGCZ?
+* How will this impact deployments?
+
+Here are the options that I considered:
+
+### [gutter-django](https://github.com/disqus/gutter-django)
+* Too complicated
+* Not actively maintained
+
+### [gargoyle-yplan](https://gargoyle-yplan.readthedocs.io/en/latest/)
+* Actively maintained fork
+* Documentation not great
+* Can add default values in `settings.py`
+* Flags are stored in database
+
+### [django-waffle](https://waffle.readthedocs.io/en/v0.11.1/)
+* Popular
+* Good documentation
+* Not very active
+* Has JavaScript interface
+* Flags are stored in database
+
+### None of the above (custom code)
+* No database required
+* We already have some conventions in place for custom settings
+
+## Decision
+
+We have decided to follow existing conventions to add custom settings
+to MMW to support BiG CZ instead of adding a feature flipper library.
+
+Because flags will rarely change, and we don't need any advanced features
```

Comment:
BiG CZ will have a custom sidebar to explore data sources from external APIs. See https://marvelapp.com/2bj76ci.